apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1171,7 +1134,7 @@ func (u *UniterAPI) EnterScope(ctx context.Context, args params.RelationUnits) (
 			return nil
 		}
 
-		netInfo, err := NewNetworkInfo(ctx, u.st, u.networkService, unitTag, u.logger)
+		netInfo, err := NewNetworkInfo(ctx, u.st, u.networkService, u.modelConfigService, unitTag, u.logger)
```

Comment:
`NetworkInfo` logic will ultimately end up behind the service boundary. 

It will be simplified and not need bifurcation for IAAS and CAAS, because we model network interfaces consistently.